gui/wxpython/gui_core/vselect.py

```diff
@@ -269,10 +269,21 @@ def GetSelectedMap(self):
             return None
 
         if layerSelected:
-            mapName = str(layerSelected)
-            if self.mapName is not None:
-                if self.mapName != mapName:
-                    self.Reset()
+            if layerSelected.type == 'raster':
```

Comment:
I think type != 'vector' would be better since there are also 3D rasters.

Reply:
Yes you are right I have already fixed it.